oap-server/server-receiver-plugin/configuration-discovery-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/recevier/configuration/discovery/AgentConfigurations.java

```diff
@@ -37,9 +41,23 @@ public class AgentConfigurations {
      */
     private volatile String uuid;
 
+    private volatile boolean needMerge = true;
+
     public AgentConfigurations(final String service, final Map<String, String> configuration, final String uuid) {
         this.service = service;
         this.configuration = configuration;
         this.uuid = uuid;
     }
+
+    public void mergeAgentConfigurations(Map<String, String> config) {
```

Comment:
Actually, you don't need this. The only thing `ConfigurationDiscoveryServiceHandler#newAgentDynamicConfigCommand` required is `getConfiguration`. You could build an iterator by including both config.

You could do `agentConfigurations.buildConfiguraion(agentConfigurationsWatcher.getAgentConfigurations("default-config"))`.`foreach`.

By this, you don't need `needMerge`, also don't need to really merge two maps.